core/pom.xml

```diff
@@ -357,6 +357,34 @@
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-crypto</artifactId>
     </dependency>
+
+    <!--
+     Testing Hive reflection needs hive on the test classpath only.
```

Comment:
We went back in forth on this in the main thread, which is admittedly long and hard to follow at this point.  Here's the same concern laid out by mridulm: https://github.com/apache/spark/pull/17723#issuecomment-298222247

We ultimately decided that it would be acceptable to add these dependencies, so long as we don't expose them in the public Spark interfaces.  Here's a comment to that effect: https://github.com/apache/spark/pull/17723#issuecomment-298463750

The reason they're required is that this PR moves the Hadoop delegation token providers from the yarn module into the core module, so that other resource managers, such as Mesos, can also fetch delegation tokens to access secure hadoop services.  One of those delegation token providers is Hive: https://github.com/apache/spark/pull/17723/files#diff-8be1059872a069d1bb4c5fc3ca394968